extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/security/SecurityContextOverrideHandler.java

```diff
@@ -0,0 +1,131 @@
+        CurrentIdentityAssociation currentIdentityAssociation = Arc.container().select(CurrentIdentityAssociation.class).get();
+        Uni<SecurityIdentity> oldIdentity = currentIdentityAssociation.getDeferredIdentity();
+        currentIdentityAssociation.setIdentity(oldIdentity.map(new Function<SecurityIdentity, SecurityIdentity>() {
+            @Override
+            public SecurityIdentity apply(SecurityIdentity old) {
+                Set<Credential> oldCredentials = old.getCredentials();
+                Map<String, Object> oldAttributes = old.getAttributes();
+                return new SecurityIdentity() {
+                    @Override
+                    public Principal getPrincipal() {
+                        return modified.getUserPrincipal();
+                    }
+
+                    @Override
+                    public boolean isAnonymous() {
+                        return modified.getUserPrincipal() == null;
+                    }
+
+                    @Override
+                    public Set<String> getRoles() {
+                        throw new UnsupportedOperationException(
+                                "retrieving all roles not supported when JAX-RS security context has been replaced");
+                    }
+
+                    @Override
+                    public boolean hasRole(String role) {
+                        return modified.isUserInRole(role);
+                    }
+
+                    @Override
+                    public <T extends Credential> T getCredential(Class<T> credentialType) {
+                        for (Credential cred : getCredentials()) {
+                            if (credentialType.isAssignableFrom(cred.getClass())) {
+                                return (T) cred;
+                            }
+                        }
+                        return null;
+                    }
+
+                    @Override
+                    public Set<Credential> getCredentials() {
+                        return oldCredentials;
+                    }
+
+                    @Override
+                    public <T> T getAttribute(String name) {
+                        return (T) oldAttributes.get(name);
+                    }
+
+                    @Override
+                    public Map<String, Object> getAttributes() {
+                        return oldAttributes;
+                    }
+
+                    @Override
+                    public Uni<Boolean> checkPermission(Permission permission) {
+                        return Uni.createFrom().nullItem();
+                    }
+                };
+            }
+        }));
+    }
+
+    private CurrentIdentityAssociation getCurrentIdentityAssociation() {
+        CurrentIdentityAssociation identityAssociation = this.currentIdentityAssociation;
+        if (identityAssociation == null) {
+            return this.currentIdentityAssociation = CDI.current().select(CurrentIdentityAssociation.class).get();
```

Comment:
Probably best to be consistent and use `Arc.container()` instead of `CDI.current()`